onnxruntime/core/providers/cpu/tensor/upsample.h

```diff
@@ -307,7 +307,24 @@ class UpsampleBase {
                                      const std::vector<int64_t>& input_dims,
                                      std::vector<float>& scales) const {
     for (size_t i = 0, end = input_dims.size(); i < end; ++i) {
-      scales[i] = static_cast<float>(output_dims[i]) / static_cast<float>(input_dims[i]);
+      // Handle corner case to avoid dividing by zero in the next step
+      if (input_dims[i] == 0) {
```

Comment:
Semi-related given this change is about handling a zero. Should Upsample<T>::BaseCompute check if the output size will be 0 after creating Y and exit early?

Reply:
Added the logic